src/AutoRest.CSharp/Common/Input/Source/CompilationInput.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.Input.Source
+{
+    public record CompilationInput(Compilation Compilation, Func<INamedTypeSymbol, bool>? typeFilter = null, Func<IMethodSymbol, bool>? methodFilter = null);
```

Comment:
nit
```suggestion
    public record CompilationInput(Compilation Compilation, Func<INamedTypeSymbol, bool>? TypeFilter = null, Func<IMethodSymbol, bool>? MethodFilter = null);
```
Also I suggest we add two methods on this record so that we do not have to do the null check when we check the symbol in other classes, something like
```
public bool CheckType(INamedTypeSymbol type) => typeFilter?.Invoke(type) ?? false;

public bool CheckMethod(IMethodSymbol method) => methodFilter?.Invoke(method) ?? false;
```